src/app/components/recipe-page/recipe-page.component.ts

```diff
@@ -1,45 +1,46 @@
 import { Component, OnInit } from '@angular/core';
-import { ApiService } from "./../api.service";
+import { ApiService } from './../api.service';
 import { Recipe } from './../models/recipe.model';
 import { ActivatedRoute, Router } from '@angular/router';
-import { switchMap } from 'rxjs/operators';
-
-
-
 
 @Component({
   selector: 'app-recipe-page',
   templateUrl: './recipe-page.component.html',
-  styleUrls: ['./recipe-page.component.css']
+  styleUrls: ['./recipe-page.component.css'],
 })
-
-
-
 export class RecipePageComponent implements OnInit {
   recipe: Recipe;
+  public edit: boolean;
   public id: Number;
-  constructor(private recipesService: ApiService,
+  constructor(
+    private recipesService: ApiService,
     private route: ActivatedRoute,
     private router: Router
-    ) {
-    // this.recipe = ;
-  }
+  ) {}
 
   ngOnInit(): void {
+    this.edit = false;
     this.id = parseInt(this.route.snapshot.paramMap.get('id'));
-    this.recipesService.getRecipe(this.id).subscribe((recipe: any) =>
-    {
-      this.recipe = recipe;
-      console.log(recipe)
-    })
+    this.getRecipe();
```

Comment:
I'd modify this so that this method returns recipe, not assigns it to the component variable. You can assign it like this:

```
this.recipe = this.getRecipe();
```